sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeBatchActionsAsyncClient.java

```diff
@@ -0,0 +1,445 @@
+                            return null;
+                        }
+                        final KeyPhrasesTask keyPhrasesTask = new KeyPhrasesTask();
+                        keyPhrasesTask.setParameters(
+                            // TODO: currently, service does not set their default values for model version, we
+                            // temporally set the default value to 'latest' until service correct it.
+                            // https://github.com/Azure/azure-sdk-for-java/issues/17625
+                            new KeyPhrasesTaskParameters()
+                                .setModelVersion(action.getModelVersion() == null
+                                                     ? "latest" : action.getModelVersion()));
+                        return keyPhrasesTask;
+                    }).collect(Collectors.toList()));
+    }
+
+    private Function<PollingContext<AnalyzeBatchActionsOperationDetail>, Mono<AnalyzeBatchActionsOperationDetail>>
+        activationOperation(Mono<AnalyzeBatchActionsOperationDetail> operationResult) {
+        return pollingContext -> {
+            try {
+                return operationResult.onErrorMap(Utility::mapToHttpResponseExceptionIfExist);
+            } catch (RuntimeException ex) {
+                return monoError(logger, ex);
+            }
+        };
+    }
+
+    private Function<PollingContext<AnalyzeBatchActionsOperationDetail>, Mono<PollResponse<AnalyzeBatchActionsOperationDetail>>>
+        pollingOperation(Function<String, Mono<Response<AnalyzeJobState>>> pollingFunction) {
+        return pollingContext -> {
+            try {
+                final PollResponse<AnalyzeBatchActionsOperationDetail> operationResultPollResponse =
+                    pollingContext.getLatestResponse();
+                // TODO: [Service-Bug] change back to UUID after service support it.
+                //  https://github.com/Azure/azure-sdk-for-java/issues/17629
+//                final UUID resultUUID = UUID.fromString(operationResultPollResponse.getValue().getResultId());
+                final String operationId = operationResultPollResponse.getValue().getOperationId();
+                return pollingFunction.apply(operationId)
+                    .flatMap(modelResponse -> processAnalyzedModelResponse(modelResponse, operationResultPollResponse))
+                    .onErrorMap(Utility::mapToHttpResponseExceptionIfExist);
+            } catch (RuntimeException ex) {
+                return monoError(logger, ex);
+            }
+        };
+    }
+
+    private Function<PollingContext<AnalyzeBatchActionsOperationDetail>, Mono<PagedFlux<AnalyzeBatchActionsResult>>>
+        fetchingOperation(Function<String, Mono<PagedFlux<AnalyzeBatchActionsResult>>> fetchingFunction) {
+        return pollingContext -> {
+            try {
+                // TODO: [Service-Bug] change back to UUID after service support it.
+                //  https://github.com/Azure/azure-sdk-for-java/issues/17629
```

Comment:
Do we have an ETA on this?

Reply:
We don't have ETA yet. Service side is working on it. 
https://msazure.visualstudio.com/Cognitive%20Services/_workitems/edit/8699611/